docs/end_to_end_acknowledgements.md

```diff
@@ -0,0 +1,63 @@
+
+## End To End Acknowledgements
+The Data Prepper framework offers a reliable way for sources to ensure the durability and reliability of data by tracking its delivery to sinks through end-to-end acknowledgements. If the pipeline has multiple sinks or multiple sub-pipelines configured, the event level acknowledgements are sent only after the event is sent to all sinks in all the sub-pipelines. The framework enables a source to create an acknowledgement set to monitor a batch of events and receive a positive acknowledgement when those events are successfully pushed to the sinks or negative acknowledgement when any one of the events could not be pushed to the sinks for any reason. In the event of a failure or crash of any component of the Data Prepper, if it is unable to send an event, the source will not receive an acknowledgement. In this case, the source will time out and can take necessary actions like retrying or logging the failure.
```

Comment:
"multiple sub-pipelines configured"

I'm not sure that sub-pipelines is quite the right term here. They are destination pipelines, not necessarily sub-pipelines.

Perhaps:

"If the pipeline has multiple sinks, including other pipeline sinks, ..."